File: frontend/src/utils/dish.js
export function isValidDishName(value) {
  const name = String(value ?? '').trim()
  return name.length >= 1 && name.length <= 50
}

export function isValidInventory(value) {
  return value !== '' && value !== null && value !== undefined
    && Number.isInteger(Number(value)) && Number(value) >= 0
}

export function buildDishPayload(form, merchantId) {
  return {
    merchantId: Number(merchantId),
    name: String(form.name ?? '').trim(),
    price: Number(form.price),
    category: String(form.category ?? '').trim() || null,
    imageUrl: String(form.imageUrl ?? '').trim() || null,
    inventory: Number(form.inventory),
  }
}

export function buildDishUpdatePayload(form) {
  const { merchantId: _merchantId, ...payload } = buildDishPayload(form, 0)
  return payload
}

File: backend/src/TakeoutPlatform.Api/Features/Merchant/Dish.cs
namespace TakeoutPlatform.Api.Features.Merchant;

public class Dish
{
    public int Id { get; set; }
    public int MerchantId { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string? Category { get; set; }
    public string? ImageUrl { get; set; }
    public int Inventory { get; set; }

    public Merchant Merchant { get; set; } = null!;
}
